module/regenerator.py

```diff
@@ -204,11 +225,46 @@ def manage_brok(self, brok):
 
     # pylint: disable=no-self-use
     def update_element(self, element, data):
-        if getattr(data, 'uuid', None):
-            element.id = data.uuid
         for prop in data:
             setattr(element, prop, data[prop])
 
+    def _set_daemon_realm(self, data):
+        """Set and return the realm the daemon is attached to
+        If no realm_name attribute exist, then use the realm attribute and set as default value All if it is empty
+        """
+        if 'realm_name' not in data:
+            if 'realm' in data:
+                if data['realm']:
+                    data['realm_name'] = data['realm']
+                else:
+                    data['realm_name'] = 'All'
```

Comment:
You car replace all the if/else by `data['realm_name'] = data['realm'] or 'All'`